edgexfoundry/edgex-go
# build(snap): stage ekuiper snap instead of local build

This PR stages a standalone [edgex-ekuiper snap](https://github.com/canonical/edgex-ekuiper-snap) into edgexfoundry snap, and it will replace the exiting kuiper app. This is to ensure backward compatibility for upgrades from 2.0/2.1 to 2.x.

- remove secure message bus setup as config files coming from edgex-ekuiper stage-snap directly
- use configure hook generating redis creds for kuiper
- update kuiper `apps` and `parts`
- update edgex-ekuiper stage-snap track `1/beta` for temporary use (replace it with `1/stable` when possible)

Signed-off-by: Mengyi Wang <mengyi.wang@canonical.com>







**If your build fails**  due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/edgex-go/blob/main/.github/Contributing.md

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] I have fully tested (add details below) this the new feature or bug fix (if not, why?)The [checkbox test ](https://github.com/canonical/edgex-checkbox-provider/blob/master/data/latest/test-rules-engine.sh) has covered the rules engine, and this PR passed it.
  <link to docs PR>

## Testing Instructions




1. Install edgex and turn on kuiper service:
```
snapcraft
snap install edgexfoundry_2.0.1-dev.71_amd64.snap --devmode
snap set edgexfoundry device-virtual=on
snap set edgexfoundry kuiper=on
```
2. Create stream and rules to ensure kuiper works:
```
# stream 1
edgexfoundry.kuiper-cli create stream stream1 '()WITH(FORMAT="JSON",TYPE="edgex")'
# rule 1 
edgexfoundry.kuiper-cli create rule rule_log '
{
   "sql":"SELECT * from stream1",
   "actions":[
      {
         "log":{
            
         }
      }
   ]
}' 
# rules 2
edgexfoundry.kuiper-cli create rule rule_edgex_message_bus '
{
   "sql":"SELECT * from stream1",
   "actions": [
      {
         "edgex": {
            "connectionSelector": "edgex.redisMsgBus",
            "topicPrefix": "edgex/events/device", 
            "messageType": "request",
            "deviceName": "device-test"
         }
      }
   ]
}'
```
Expect:
```
edgexfoundry.kuiper-cli getstatus rule rule_log
```
```
Connecting to 127.0.0.1:20498... 
{
  "source_stream1_0_records_in_total": 214983,
  "source_stream1_0_records_out_total": 214983,
  "source_stream1_0_exceptions_total": 0,
  "source_stream1_0_process_latency_us": 0,
  "source_stream1_0_buffer_length": 1,
  "source_stream1_0_last_invocation": "2022-03-17T15:09:30.757436",
  "op_2_project_0_records_in_total": 214982,
  "op_2_project_0_records_out_total": 214981,
  "op_2_project_0_exceptions_total": 0,
  "op_2_project_0_process_latency_us": 0,
  "op_2_project_0_buffer_length": 0,
  "op_2_project_0_last_invocation": "2022-03-17T15:09:30.757457",
  "sink_log_0_0_records_in_total": 214981,
  "sink_log_0_0_records_out_total": 214981,
  "sink_log_0_0_exceptions_total": 0,
  "sink_log_0_0_process_latency_us": 10,
  "sink_log_0_0_buffer_length": 0,
  "sink_log_0_0_last_invocation": "2022-03-17T15:09:30.757269"
}
```
```
edgexfoundry.kuiper-cli getstatus rule rule_edgex_message_bus
```
```
Connecting to 127.0.0.1:20498... 
{
  "source_stream1_0_records_in_total": 928309,
  "source_stream1_0_records_out_total": 928309,
  "source_stream1_0_exceptions_total": 0,
  "source_stream1_0_process_latency_us": 0,
  "source_stream1_0_buffer_length": 0,
  "source_stream1_0_last_invocation": "2022-03-17T15:10:17.534734",
  "op_2_project_0_records_in_total": 928309,
  "op_2_project_0_records_out_total": 928309,
  "op_2_project_0_exceptions_total": 0,
  "op_2_project_0_process_latency_us": 2,
  "op_2_project_0_buffer_length": 0,
  "op_2_project_0_last_invocation": "2022-03-17T15:10:17.534737",
  "sink_edgex_0_0_records_in_total": 928267,
  "sink_edgex_0_0_records_out_total": 928266,
  "sink_edgex_0_0_exceptions_total": 0,
  "sink_edgex_0_0_process_latency_us": 52,
  "sink_edgex_0_0_buffer_length": 40,
  "sink_edgex_0_0_last_invocation": "2022-03-17T15:10:17.534711"
}
```
```
edgexfoundry.kuiper-cli show rules
```
```
[
  {
    "id": "rule_edgex_message_bus",
    "status": "Running"
  },
  {
    "id": "rule_log",
    "status": "Running"
  }
]
```



## New Dependency Instructions (If applicable)
